app/main.py

```diff
@@ -1,2 +1,56 @@
+import math
+
+
 class Dictionary:
-    pass
+    RESIZE = 2
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self, capacity=8):
+        self.capacity = capacity
+        self.ls = [None for _ in range(self.capacity)]
+        self.filled = 0
+
+    def __setitem__(self, key, value):
+        if self.filled == math.ceil(self.capacity * self.LOAD_FACTOR):
+            self.resize()
+
+        hash_value = hash(key)
+        index = hash_value % self.capacity
+
+        if self.ls[index] is None:
+            self.ls[index] = [key, hash_value, value]
+            self.filled += 1
+        else:
+            if key == self.ls[index][0]:
+                self.ls[index][2] = value
+                return
+            while True:
+                index += 1
+                index %= self.capacity
+                if self.ls[index] is None:
+                    self.filled += 1
+                    self.ls[index] = [key, hash_value, value]
+                    break
```

Comment:
Tests fails in 20% times with your solutions. You didn't check if cell has the same key in this loop.